parsec/parsec_mpi_funnelled.c

```diff
@@ -407,6 +407,49 @@ int parsec_mpi_sendrecv(parsec_comm_engine_t *ce,
     return (MPI_SUCCESS == rc ? 0 : -1);
 }
 
+/**
+ * Store the user provided communicator in the PaRSEC context. We need to make a
+ * copy to make sure the communicator does not disappear before the communication
+ * engine starts up.
+ * 
+ * This function is collective, all processes in the current and the new communicator
+ * should call it in same time.
+ */
+static int parsec_mpi_set_ctx(parsec_comm_engine_t* ce, intptr_t opaque_comm_ctx )
+{
+    parsec_context_t* context = ce->parsec_context;
+    MPI_Comm comm;
+    int rc;
+
+    /* We can only change the communicator if the communication engine is not active */
+    if( 1 < parsec_communication_engine_up ) {
+        parsec_warning("Cannot change PaRSEC's MPI communicator while the engine is running [ignored]");
+        return PARSEC_ERROR;
+    }
+
+    /* The engine was never yet started, so there is nothing to clean */
+    if( NULL != ce->sync ) {
+        ce->disable(ce);
+        ce->sync(ce);
+        assert( -1 != context->comm_ctx );
+        MPI_Comm_free((MPI_Comm*)&context->comm_ctx);
+    }
+    rc = MPI_Comm_dup((MPI_Comm)opaque_comm_ctx, &comm);
```

Comment:
Should we `MPI_Comm_dup_with_info` here and set the `mpi_assert_allow_overtaking` info key?

Reply:
We don't need to do that here, we never use this communicator for AM, so we don't really need to allow overtake on it. But we can create the rest of the communicators using the `dup_with_info`.